File: 8-kyu-kata/function-3-multiplying-two-numbers/python/function_3_multiplying_two_numbers_test.py
# 8 kyu - Function 3 - multiplying two numbers  [ ID: 523b66342d0c301ae400003b  (function-3-multiplying-two-numbers) ]
# URL: https://www.codewars.com/kata/523b66342d0c301ae400003b
# Category: REFERENCE  |  Tags: FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import multiply


@test.describe("Fixed Tests")
def fixed_tests():

    from inspect import isfunction

    @test.it("Should have function 'multiply'")
    def function_multiply():
        test.expect(isfunction(multiply) == True, "'multiply' is not defined as function")

    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(multiply(2, 3), 6)


@test.describe("Random Tests")
def random_tests():

    from random import randint

    for i in range(100):
        x, y = randint(0, 100), randint(1, 200)

        @test.it(f"testing for multiply({x}, {y})")
        def test_case():
            test.assert_equals(multiply(x, y), x * y)

File: 8-kyu-kata/function-3-multiplying-two-numbers/javascript/function3MultiplyingTwoNumbers.Test.js
// 8 kyu - Function 3 - multiplying two numbers  [ ID: 523b66342d0c301ae400003b  (function-3-multiplying-two-numbers) ]
// URL: https://www.codewars.com/kata/523b66342d0c301ae400003b
// Category: REFERENCE  |  Tags: FUNDAMENTALS
// *****************************************************************************
const chai = require("chai")
const assert = chai.assert

function getRandom() {
  return (Math.random() * 1000 - 500) | 0
}

describe("Testing multiply", function () {
  it("Should return 10 multiplications correctly", function () {
    for (let i = 0; i < 10; i++) {
      let x = getRandom(),
        y = getRandom()
      assert.strictEqual(multiply(x, y), x * y, "that's not the correct multiplication!")
    }
  })
})
